src/traces/surface/attributes.js

```diff
@@ -44,6 +44,62 @@ function makeContourAttr(axLetter) {
                 'dimension are drawn.'
             ].join(' ')
         },
+        start: {
+            valType: 'number',
+            dflt: null,
+            role: 'style',
+            editType: 'plot',
+         // impliedEdits: {'^autocontour': false},
+            description: [
+                'Sets the starting contour level value.',
+                'Must be less than `contours.end`'
+            ].join(' ')
+        },
+        end: {
+            valType: 'number',
+            dflt: null,
+            role: 'style',
+            editType: 'plot',
+         // impliedEdits: {'^autocontour': false},
+            description: [
+                'Sets the end contour level value.',
+                'Must be more than `contours.start`'
+            ].join(' ')
+        },
+        size: {
+            valType: 'number',
+            dflt: null,
+            min: 0,
+            role: 'style',
+            editType: 'plot',
+         // impliedEdits: {'^autocontour': false},
+            description: [
+                'Sets the step between each contour level.',
+                'Must be positive.'
+            ].join(' ')
+        },
+        locations: {
```

Comment:
Do we really need `locations` and `relative` to :white_check_mark: this feature?

I would prefer making `surface` on-par with `contour` to start.

Reply:
Good call. Let's put this on 1.47.0.

Reply:
Dropped those features in https://github.com/plotly/plotly.js/pull/3469/commits/98aebc52f9d56eee09a95828d851f75395903e7c as `surface` code is already pretty complicated for us to maintain.